### ISSUE TITLE
Wrong steps and number of points in simulation dialogs after #1054

### ISSUE THREAD

**Issue by ivandi69** [ra3xdh/qucs_s#1077]
This transient simulation setup generates wrong netlist with a negative tran step: .tran -0.001 0.001 0
![image](https://github.com/user-attachments/assets/5db413ba-c27f-416b-b007-05966f1d8bf3)
![image](https://github.com/user-attachments/assets/47f4441d-3c5e-4f5e-b72c-571310af29f4)
AC linear sweep gives wrong number of points ex: 500.95:
![image](https://github.com/user-attachments/assets/20f554b5-e286-479a-8af9-5ad6ace4bdfd)
Most likely all sweeps are broken. They worked fine since #615

**Comment by ra3xdh** [comment]
@iwbnwif , Please have a look. Revert an old behavior implemented in #615 if necessary.

**Comment by iwbnwif** [comment]
Thanks for reporting, I will look into this.

**Comment by iwbnwif** [comment]
I think this is fixed by #1079, at least for the two reported issues. As far as I can tell, the behavior for both log sweeps and lin sweeps is essentially similar to the production 24.4.1 version; however, there may be some edge cases where the user inputs invalid values for log sweeps.

Also, I notice that the list 'sweeps' do not work properly in 24.4.1 or the new component dialog. Those problems are not addressed by this fix and I will open a separate issue after more investigation.
